Tests/Realm.Tests/SourceGeneration/TestClasses/ManuallyGeneratedClass.cs

```diff
@@ -0,0 +1,270 @@
+            set => _accessor.IntValue = value;
+        }
+
+        public IList<int> ListValue => _accessor.ListValue;
+
+        [MapTo("_string")]
+        public string StringValue
+        {
+            get => _accessor.StringValue;
+            set => _accessor.StringValue = value;
+        }
+
+        [PrimaryKey]
+        public int PrimaryKeyValue
+        {
+            get => _accessor.PrimaryKeyValue;
+            set => _accessor.PrimaryKeyValue = value;
+        }
+
+        #endregion
+    }
+
+    [Woven(typeof(ManuallyGeneratedClassObjectHelper))]
+    public partial class ManualllyGeneratedClass : IRealmObject, INotifyPropertyChanged
+    {
+        // The schema property could be part of an interface, but for that the user needs to have at least .NET 5.0 and C# 8.0
+        // So we need to use reflection to check if this exists.
+        public static ObjectSchema RealmSchema = new ObjectSchema.Builder("ManualllyGeneratedClass", isEmbedded: false)
+        {
+            Property.Primitive("IntValue", RealmValueType.Int),
+            Property.PrimitiveList("ListValue", RealmValueType.Int),
+            Property.Primitive("_string", RealmValueType.String),
+            Property.Primitive("PrimaryKeyValue", RealmValueType.Int, isPrimaryKey: true),
+        }.Build();
+
+        #region IRealmObject implementation
+
+        private IManuallyGeneratedClassAccessor _accessor;
+
+        public IRealmAccessor Accessor => _accessor;
+
+        public bool IsManaged => _accessor.IsManaged;
+
+        public bool IsValid => _accessor.IsValid;
+
+        public bool IsFrozen => _accessor.IsFrozen;
+
+        public Realm Realm => _accessor.Realm;
+
+        public ObjectSchema ObjectSchema => _accessor.ObjectSchema;
+
+        public ManualllyGeneratedClass()
+        {
+            _accessor = new ManuallyGeneratedClassUnmanagedAccessor(typeof(ManualllyGeneratedClass));
+        }
+
+        public void SetManagedAccessor(IRealmAccessor managedAccessor, IRealmObjectHelper helper = null, bool update = false, bool skipDefaults = false)
+        {
+            var unmanagedAccessor = _accessor;
+            _accessor = (IManuallyGeneratedClassAccessor)managedAccessor;
+
+            if (helper != null)
+            {
+                IntValue = unmanagedAccessor.IntValue;
+                StringValue = unmanagedAccessor.StringValue;
+
+                if (!skipDefaults)
+                {
+                    ListValue.Clear();
+                }
+
+                foreach (var val in unmanagedAccessor.ListValue)
+                {
+                    ListValue.Add(val);
+                }
+            }
+
+            if (_propertyChanged != null)
+            {
+                SubscribeForNotifications();
+            }
+
+            OnManaged();
+        }
+        #endregion
+
+        private event PropertyChangedEventHandler _propertyChanged;
+
+        public event PropertyChangedEventHandler PropertyChanged
```

Comment:
I'm not sure I see how unmanaged object notifications are wired up.

Reply:
Actually for managed as well - should we be raising propertychanged every time a property is set?

Reply:
I really thought notifications for unmanaged objects didn't work... my bad. I'm adding more tests to the manual generation test fixture. 

For managed that's done in the same way as it was done before. After `SubscribeToNotifications` is called, then `RaisePropertyChanged` it's called from native every time it's needed.